code/client/src/Client.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Client.h>
+
+bool Client::updateModule(){
+//    sendRequest();
+    return true;
+}
+
+void Client::sendRequest() {
+
+    yarp::os::Bottle outputBottle;
+    yarp::os::Bottle inputBottle;
+    outputBottle.addVocab(COLLATZ_VOCAB_REQ_ITEM);
+    outputBottle.addInt(n);
+    if (testOutcome == 0) {
+        std::cout << "Client side: Test unsuccessful. Sending 0 request" << std::endl;
+    }
+    std::cout << "Sending request " << n << std::endl;
+    clientPort.write(outputBottle,inputBottle);
+
+    if (inputBottle.get(0).asVocab() != COLLATZ_VOCAB_ITEM) {
+        std::cerr << "Client side: Incorrect Identifier received. Ignoring message," << std::endl;
+        return;
+    }
+    n = inputBottle.get(1).asInt();
```

Comment:
You don't check the size of the received Bottle; what happens if `inputBottle.size()==1`?